### DISCUSSION TITLE
How to deploy one specific branch without default branch?

### DISCUSSION THREAD

**Discussion by itsmnthn** [vercel/community#1185] [Help] [answered]
### Summary

I have a GitHub repo that has two branches and needs three deployments
1. `v2` is the default branch(v2 production diff environment var) and has the latest commits beta testing changes and serves at `beta.acme.com`
2. `main` is the second branch(v1 production for b2b diff environment var) and has the live app and serves at `app.acme.com`
3. `main` is the second branch(v1 production for b2c diff environment var) and has the live app and serves at `some.acme.com`

Using Netlify as of now but wanna switch to Vercel and I find it hard to deploy these three static apps on three separate project

1. I find it easy to deploy `v2` the default branch 
2. but when I try to deploy the `main` branch it works but if I push changes to `v2` branch then deployment triggers on this project for `v2` which is not ideal and do not know how to prevent this (note: I have set `main` branch as production in project settings)
3. happens the same as point 2

**Comment by amyegan** [comment, marked as answer]
Hi @itsmnthn! It sounds like you're trying to deploy each branch to it's own separate project but ignore the other branches. The [Ignored Build Step](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel) would be good for this. The setting can be found in the Git section of your project settings.

In your "v2" project, you could use `if [ "$VERCEL_GIT_COMMIT_REF" != "v2" ]; then exit 1; else exit 0; fi` to skip deployments for any branch that is not "v2". 

Or you could use `if [ "$VERCEL_GIT_COMMIT_REF" == "main" ]; then exit 1; else exit 0; fi` to skip all deployments for commits to "main", but allow deployments as normal from "v2" and other branches.

Whichever you choose, be sure to use a similar setting for your "main" branch projects to skip unwanted deployments there as well!

**Reply by itsmnthn** [reply]
I think this should be a good/helpful feature to have a UI inside the git section in settings. can be included in the flow of creating a project to avoid the first unwanted build.

**Reply by amyegan** [reply]
Good suggestion! I'm adding your feedback to the requested features list 🙏

**Reply by itsmnthn** [reply]
is this requested feature list open to the public? I would like to add a few suggestions to improve overall UX.

**Reply by itsmnthn** [reply]
v2 branch project
```bash
[10:37:00.411] Cloning github.com/username/repo (Branch: v2, Commit: 3809148)
[10:37:00.419] Skipping build cache since Vercel CLI used --force option
[10:37:02.389] Cloning completed: 1.977s
[10:37:02.389] Running "echo $VERCEL_GIT_COMMIT_REF; if [ "$VERCEL_GIT_COMMIT_REF" != "v2" ]; then exit 1; else exit 0; fi"
[10:37:02.500] v2
[10:37:02.502] The Deployment has been canceled as a result of running the command defined in the "Ignored Build Step" setting.
```

**Reply by itsmnthn** [reply]
but it seems need to switch the `exit 0` with `exit 1`

this doesn't make sense `exit 1` should ignore the build and `exit 0` should continue but happens the other way

main branch project
```bash
[10:45:29.261] Cloning github.com/username/repo (Branch: v2, Commit: 86712e7)
[10:45:31.312] Cloning completed: 2.050s
[10:45:31.312] Running "echo $VERCEL_GIT_COMMIT_REF; if [ "$VERCEL_GIT_COMMIT_REF" != "main" ]; then exit 0; else exit 1; fi"
[10:45:31.441] v2
[10:45:31.442] The Deployment has been canceled as a result of running the command defined in the "Ignored Build Step" setting.
```

**Reply by amyegan** [reply]
Ah, you're right. Looks like that example I pulled from docs is backwards. I'll get that guide updated. Sorry for my mixup 🙇 

These discussion posts are the public place to share ideas and feature requests, so please comment here or [start a new discussion](https://github.com/orgs/vercel/discussions/new?category=ideas) with any ideas you have. I'll make sure they are shared with the rest of the team!
